catlin/pkg/linter/script.go

```diff
@@ -0,0 +1,171 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linter
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/plumbing/catlin/pkg/parser"
+	"github.com/tektoncd/plumbing/catlin/pkg/validator"
+)
+
+type taskLinter struct {
+	res     *parser.Resource
+	configs []config
+}
+
+type linter struct {
+	cmd  string
+	args []string
+}
+
+type config struct {
+	regexp  string
+	linters []linter
+}
+
+// NewConfig construct default config
+func NewConfig() []config {
+	return []config{
```

Comment:
For a script with no shebang `#!` we should also run shellcheck and `/bin/sh -n` 🙃

Reply:
yeah good point, i have added it in the tests but forgot to bring it back to the main config,

Reply:
nm my previous comment i do have it: 

https://github.com/tektoncd/plumbing/pull/679/files/0c8f582d16de69ab65fe52c44d7b100d0339c8ef#diff-bf727d57cc66d19dbee8015f9adcddab76e0d36677f82b2dd348ca237ec7e181R89-R92

Reply:
ah right 😅